content/automate/ManageIQ/Transformation/Infrastructure/VM/Common.class/__methods__/restorevmattributes.rb

```diff
@@ -0,0 +1,46 @@
+module ManageIQ
+  module Automate
+    module Transformation
+      module Infrastructure
+        module VM
+          module Common
+            class RestoreVmAttributes
+              def initialize(handle = $evm)
+                @handle = handle
+              end
+
+              def main
+                task = @handle.root['service_template_transformation_plan_task']
+                source_vm = task.source
+                destination_vm = @handle.vmdb(:vm).find_by(:id => task.get_option(:destination_vm_id))
+                @handle.log(:info, "VM: #{destination_vm.inspect}")
+                @handle.log(:info, "VM Id: #{destination_vm.id}")
+
+                # Reconnect destination VM to service
+                destination_vm.add_to_service(source_vm.service) unless source_vm.service.nil?
+                source_vm.remove_from_service
```

Comment:
@fdupont-redhat The `remove_from_service` is a needless call if we have already determined that the source_vm is not in a service.  I would suggest turning this into an if-block which encompasses both calls.

```ruby
if source_vm.service
  destination_vm.add_to_service(source_vm.service)
  source_vm.remove_from_service
end
```

Reply:
Agreed. And thanks for the snippet. I've updated the PR.